### PR TITLE
Check lang keys at compile time

### PR DESCRIPTION
**Pull request by Librazy**
And fixed some erroneous lang keys found by it.

**Examples (in my modified source for testing):**
```
C:\Users\Liqueur Librazy\Documents\Github\nyaautils\src\main\java\cat\nyaa\nyaautils\lootprotect\LootProtectMode.java:7: 警告: Key component suffix LANG_CHECKER not found in lang en_US.yml
public enum LootProtectMode {
       ^
C:\Users\Liqueur Librazy\Documents\Github\nyaautils\src\main\java\cat\nyaa\nyaautils\CommandHandler.java:229: 警告: Key prefix user.p.mode_ not found in lang en_US.yml
        p.sendMessage(I18n.format("user.p.mode_" + plugin.cfg.lootProtectMode.name()));
                                  ^
C:\Users\Liqueur Librazy\Documents\Github\nyaautils\src\main\java\cat\nyaa\nyaautils\CommandHandler.java:365: 警告: Corrupted key user.rename.name_too_long in en_US.yml: Name '%s' is too long! Should less than %d chars
            msg(p, "user.rename.name_too_long", name);
                   ^
C:\Users\Liqueur Librazy\Documents\Github\nyaautils\src\main\java\cat\nyaa\nyaautils\CommandHandler.java:370: 警告: Key nu.warn.blocked_format_codes not found in lang en_US.yml
                msg(p, "nu.warn.blocked_format_codes", "&" + k);
                       ^
```
**Integrating with idea:**

image

image

### CODE REVIEW

**Comment by RecursiveG** [comment]
Could you explain how the annotation works on Enum types and
what does the `SUFFIX` mean?

File: src/main/java/cat/nyaa/utils/CommandReceiver.java [hunk 1]
```diff
@@ -198,7 +200,7 @@ public void acceptCommand(CommandSender sender, Arguments cmd) {
         } catch (NoItemInHandException ex) {
             msg(sender, ex.isOffHand ? "internal.error.no_item_offhand" : "internal.error.no_item_hand");
         } catch (BadCommandException ex) {
-            String msg = ex.getMessage();
+            @LangKey String msg = ex.getMessage();
```
Comment (RecursiveG) [inline src/main/java/cat/nyaa/utils/CommandReceiver.java]: How this works?
Doing compile time check on a runtime variable doesn't make sense.

Comment (Librazy) [inline src/main/java/cat/nyaa/utils/CommandReceiver.java]: Hmm... Seems that I missed something. It
 should complain that ex::getMessage havn't been annotated. It works like a data type here.
I havn't implement check on function return though

**Comment by Librazy** [comment]
For enum types, it checks their enum constants' name as a string.
For something annotated as `SUFFIX`, it should appear as a suffix. Thus, there should be a key that `endsWith` it.
For `PREFIX`, should be a key that `startsWith` it.
For `INFIX`, should be a key that `contains` it.
For `KEY`, should be a key that `equals` it.

**Pull request merged**